devices/src/acpi.rs

```diff
@@ -0,0 +1,47 @@
+// Copyright © 2019 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use vmm_sys_util::eventfd::EventFd;
+use BusDevice;
+
+/// A device for handling ACPI shutdown and reboot
+pub struct AcpiDevice {
+    exit_evt: EventFd,
+    reset_evt: EventFd,
+}
+
+impl AcpiDevice {
+    /// Constructs a device that will signal the given event when the guest requests it.
+    pub fn new(exit_evt: EventFd, reset_evt: EventFd) -> AcpiDevice {
+        AcpiDevice {
+            exit_evt,
+            reset_evt,
+        }
+    }
+}
+
+// Same I/O port used for shutdown and reboot
+impl BusDevice for AcpiDevice {
+    fn read(&mut self, _base: u64, _offset: u64, _data: &mut [u8]) {}
```

Comment:
Should we log some warning here since we don't expect this port to be read?

Reply:
I actually made it return 0 per the spec.

Reply:
Ok!